GitUI/UserControls/FileStatusList.Designer.cs

```diff
@@ -83,18 +82,6 @@ private void InitializeComponent()
             NoFiles.Margin = new Padding(0);
             NoFiles.Name = "NoFiles";
             NoFiles.Size = new Size(65, 13);
-            NoFiles.TabIndex = 5;
-            // 
-            // LoadingFiles
-            // 
-            LoadingFiles.AutoSize = true;
-            LoadingFiles.BackColor = SystemColors.Window;
-            LoadingFiles.ForeColor = SystemColors.GrayText;
-            LoadingFiles.Location = new Point(4, 21);
-            LoadingFiles.Margin = new Padding(0);
-            LoadingFiles.Name = "LoadingFiles";
-            LoadingFiles.Size = new Size(65, 13);
-            LoadingFiles.TabIndex = 5;
```

Comment:
Loading can take long. I would not like keeping the old file list during reload. It is not clear whether it will be updated yet or not.
If you want to avoid flicker on quick loading, you could delay the display of "Loading data..." for 300 ms.

Reply:
> I would not like keeping the old file list during reload.

The list is cleared when updating.

> If you want to avoid flicker on quick loading, you could delay the display of "Loading data..." for 300 ms.

It is not many updates that requires more than 300 ms.
I considered to add with a delay, but would like to avoid the complexity here, suggest the label is removed.

But will keep the filter box when loading.

Reply:
> It is not many updates that requires more than 300 ms.

Maybe in your repos - but not in some at work and in the Linux repo (while it is loading *all* commits yet, also afterwards on cursor down - for many seconds sometimes). Updating the Commit Info might be delaying the FileStatusList update.

Reply:
I have tested with other repos, including Linux.
I do not see the *Files Loading* label as important, a blank list is clear to me, it is not often iy appears.

I implemented this though. Some of the complexity is due to always use the cancellation token which is not bad, other I would have preferred to avoid.

Reply:
There seems to be something blocking the UI thread.
"Loading data..." should appear after at most 300 ms.

When testing, I even saw no list update for more than a second. The cleared list was noticeable more often than "Loading data..."
(The machines used for testing are able to rebuild GE in less than a minute, i.e. are not slow.)

I think the Commit Info tasks need to be cancelled, too:

![image](https://github.com/gitextensions/gitextensions/assets/36601201/07fba463-9a85-4f84-9c5b-006e64697fc9)

without "contained in / derived from tag":

![image](https://github.com/gitextensions/gitextensions/assets/36601201/91ad2d5c-b31c-4603-adfe-f071f7a3a3bc)

---

The diff group is missing if there is only one. This is annoying and breaks "Find" (search in files of diff).

![image](https://github.com/gitextensions/gitextensions/assets/36601201/3d643757-54d3-4cc8-b1f8-a2d9d2baad90)

Reply:
> There seems to be something blocking the UI thread. "Loading data..." should appear after at most 300 ms.
> 
> When testing, I even saw no list update for more than a second. The cleared list was noticeable more often than "Loading data..." (The machines used for testing are able to rebuild GE in less than a minute, i.e. are not slow.)
> 
> I think the Commit Info tasks need to be cancelled, too:

I do not see this delay myself.
Agree for CommitInfo, separate though.

> The diff group is missing if there is only one. This is annoying and breaks "Find" (search in files of diff).

8798a0524a16081da452168d61e219f9b76ee1e0

Reply:
just cursor down on the (loaded) Linux repo (Commit Info without tags):

![filestatuslist_async](https://github.com/gitextensions/gitextensions/assets/36601201/1d00c7af-4e3f-4d51-8e96-72932c082f77)

> Agree for CommitInfo, separate though.
>
> https://github.com/gitextensions/gitextensions/commit/8798a0524a16081da452168d61e219f9b76ee1e0

👍 both

Reply:
Try https://github.com/gerhardol/gitextensions/pull/new/feature/commitinfo-cancel-token 959cb0de6bc059ab0aa16b125d71d2f738a5de3c
for cancelling CommitInfo
(maybe adding for git-describe too, just 200 ms though
Still a separate PR